src/idpi/operators/destagger.py

```diff
@@ -84,42 +84,52 @@ def destagger(
     ----------
     field : xr.DataArray
         Field to destagger
-    dim : Literal["x", "y", "generalVertical"]
+    dim : Literal["x", "y", "z"]
         Dimension along which to destagger
 
     Raises
     ------
     ValueError
         Raises ValueError if dim argument is not one of
-        {"x","y","generalVerticalLayer"}.
+        {"x","y","z"}.
 
     Returns
     -------
     xr.DataArray
         destaggered field with dimensions in
-        {"x","y","generalVerticalLayer"}
+        {"x","y","z"}
 
     """
     dims = list(field.sizes.keys())
     if dim == "x" or dim == "y":
-        return xr.apply_ufunc(
-            interpolate_midpoint,
-            field.reset_coords(drop=True),
-            input_core_dims=[[dim]],
-            output_core_dims=[[dim]],
-            kwargs={"extend": "left"},
-        ).transpose(*dims)
-    elif dim == "generalVertical":
+        if field.origin[dim] != 0.5:
```

Comment:
we could extract the argument consistency checks from here to a separate call. That way it would be easier to extend and turn off if we want to

Reply:
The consistency checks definitely need to be extended at some point for all operators but this requires some additional thought regarding the design. That said, I don't understand the suggestion that you are making. Should it be a call that is done before the call to destagger? How would you ensure that it has been called from the point of view of this function?

Reply:
I would do 
```
def destagger(...):
  check_field() -> checks on x,y and z based on the input, raise if none of those
  if x,y:
    apply_ufunc1
  elif z:
   appy_ufunc2
```

Reply:
I see thanks for the snippet. The condition to check is that origin = 0.5 for x and y, and origin = -0.5 for z. The check would therefore have a similar structure to the current function but with the added indirection. I think this layout would make sense if there were more checks that are in common for all directions.